libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetryTest.kt

```diff
@@ -342,6 +344,80 @@ class MapboxNavigationTelemetryTest {
         assertEquals(10, events.size)
     }
 
+    @Test
+    fun cache_feedback_is_cached() {
+        baseMock()
+        mockAnotherRoute()
+
+        baseInitialization()
+        val cacheFeedbackType = FeedbackEvent.POSITIONING_ISSUE
+        cacheUserFeedback(feedbackType = cacheFeedbackType)
+        val cacheFeedbackDescription = "cacheFeedbackDescription"
+        cacheUserFeedback(description = cacheFeedbackDescription)
+        updateSessionState(FREE_DRIVE)
+
+        val cachedFeedbackEvents = MapboxNavigationTelemetry.getCachedUserFeedback()
+        assertEquals(cachedFeedbackEvents.size, 2)
+        assertEquals(
+            cacheFeedbackType,
+            cachedFeedbackEvents[0].feedbackType
+        )
+        assertEquals(
+            "cacheFeedbackDescription",
+            cachedFeedbackEvents[1].description
+        )
+    }
+
+    @Test
+    fun cache_feedback_not_send_on_session_stop() {
+        baseMock()
+        mockAnotherRoute()
+
+        baseInitialization()
+        cacheUserFeedback()
+        offRoute()
+        updateRoute(anotherRoute)
+        updateRouteProgress()
+        postUserFeedback()
+        cacheUserFeedback()
+        updateSessionState(FREE_DRIVE)
+
+        val events = captureAndVerifyMetricsReporter(exactly = 5)
+        assertTrue(events[0] is NavigationAppUserTurnstileEvent)
+        assertTrue(events[1] is NavigationDepartEvent)
+        assertTrue(events[2] is NavigationRerouteEvent)
+        assertTrue(events[3] is NavigationFeedbackEvent)
+        assertTrue(events[4] is NavigationCancelEvent)
+        assertEquals(5, events.size)
+    }
+
+    @Test
+    fun cache_feedback_sent_even_after_session_stop() {
+        baseMock()
+        mockAnotherRoute()
+
+        baseInitialization()
+        cacheUserFeedback()
+        offRoute()
+        updateRoute(anotherRoute)
+        updateRouteProgress()
+        postUserFeedback()
+        cacheUserFeedback()
+        updateSessionState(FREE_DRIVE)
+        postCachedUserFeedback()
```

Comment:
just to clarify
right now `cancel` event is always a final event in each session.
it means that on `cancel` event we call `locationsCollector.flushBuffers` and we immediately post all pending events (which are waiting for location buffer collection).

so, cachedEvents might be sent `after` `cancel`. 
@Guardiola31337

Reply:
correct.